### PR TITLE
Parser/empik

### PR DESCRIPTION
**Pull request by sledzgrzegorz**
This PR contains parser for empik ebook promotions.

### CODE REVIEW

File: parser/src/test/java/parser/empik/Utils.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        document.setBaseUri(pageUrl);
+        return document;
+    }
+
+    public static List<String> loadFileByLineToList(String relativePath)  {
+        List<String> textLineByLine = new ArrayList<>();
+
+        try (Stream<String> stream = Files.lines(getPathInResources(relativePath))) {
+            textLineByLine = stream.collect(Collectors.toList());
+        } catch (IOException e) {
```
Comment (caroo7) [inline parser/src/test/java/parser/empik/Utils.java]: Change to one 'catch' block.

File: parser/src/main/java/parser/PageLoader.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            Connection.Response response = Jsoup.connect(requestUrl)
+                    .userAgent(userAgent)
+                    .method(Connection.Method.GET)
+                    .timeout(0)
+                    .execute();
+            document = response.parse();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return document;
```
Comment (tdudzik) [inline parser/src/main/java/parser/PageLoader.java]: Returned value can be null here. Are you sure that you're handling this in a proper way in places you use this method? You can use Optional for example.

File: parser/src/test/java/parser/empik/EmpikBookCrawlerTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+        //act
+        List<Book> books=bookCrawler.getBooks(urlsToBookDetails);
+        Book resultBook=books.get(0);
+        //assert
+        SoftAssert sf=new SoftAssert();
```
Comment (tdudzik) [inline parser/src/test/java/parser/empik/EmpikBookCrawlerTest.java:51]: :+1:

File: parser/src/main/java/parser/BookParser.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package parser;
+
+import org.jsoup.nodes.Document;
+import parser.DTO.Book;
+
+/**
+ * Created by Grzesiek on 2016-08-28.
+ */
+public abstract class BookParser implements IBookParser {
```
Comment (tdudzik) [inline parser/src/main/java/parser/BookParser.java]: Is there a need to implement IBookParser here? Are you going to have more than one class that implements this interface? BookParser is an abstract class, maybe it is enough?

**Comment by tdudzik** [comment]
There is some things to change but overall I like the code. Maybe a little of docs would be nice because there are a lot of classes, interfaces, etc. and with that it would be easier to understand.

I've also run this code on my machine and it works (and it works quite fast).

File: parser/src/main/java/parser/PageLoader.java (hunk 2 above)
Comment (sledzgrzegorz) [reply parser/src/main/java/parser/PageLoader.java]: fixed :)

File: parser/src/test/java/parser/empik/Utils.java (hunk 1 above)
Comment (sledzgrzegorz) [reply parser/src/test/java/parser/empik/Utils.java]: fixed :)

File: parser/src/main/java/parser/BookParser.java (hunk 4 above)
Comment (sledzgrzegorz) [reply parser/src/main/java/parser/BookParser.java]: You're right. I left only interface. It is fixed now.

**Pull request merged**